### ISSUE TITLE
Unacceptably slow

### ISSUE THREAD

**Issue by Alexander-Shukaev** [alpaker/fill-column-indicator#87]
With buffer size of 53 MB, jumping from bottom to top hangs Emacs (forever?).  Doing `<C-g>` and printing `profiler-report`, shows the following offenders:

```
- redisplay_internal (C function)                               59351  56%
 - fci-update-window-for-scroll                                 59285  56%
  - fci-redraw-window                                           59285  56%
   - fci-redraw-region                                          59283  56%
      fci-put-overlays-region                                   58367  55%
- command-execute                                               43059  40%
 - call-interactively                                           43059  40%
  - funcall-interactively                                       43059  40%
   - counsel-M-x                                                38137  36%
    - ivy-read                                                  37904  36%
     - read-from-minibuffer                                     37593  35%
      - #<compiled 0x11f4131>                                   36140  34%
       - ivy--minibuffer-setup                                  36139  34%
        - ivy--exhibit                                          36139  34%
         - ivy--insert-minibuffer                               36104  34%
          - ivy--resize-minibuffer-to-fit                       36104  34%
           - window-resize                                      36103  34%
            - window--resize-mini-window                        36103  34%
             - run-window-configuration-change-hook             36103  34%
              - fci-redraw-frame                                36102  34%
               - fci-delete-unneeded                            36040  34%
```

I've been using `fci-mode` for years but this is unacceptable and now goes straight to `:disabled` until the performance issue is fixed.

**Comment by alpaker** [comment]
Sorry, this package is no longer actively maintained.

**Comment by Alexander-Shukaev** [comment]
I noticed that already and I opened this for documentation purposes in the first place.

**Comment by alpaker** [comment]
The functionality provided by this package has now been implemented natively in Emacs as `display-fill-column-indicator-mode`, available as of version 27.0.90. Unless you are forced to use an older Emacs, you should use the native implementation instead of `fci-mode`. The former is unaffected by most of the compatibility issues that this add-on has.
